File: easy/3314/main.cpp
class Solution {
public:
    vector<int> minBitwiseArray(vector<int>& nums) {
      vector<int> ans;

      for (int num : nums) {
        ans.push_back(-1);
        for (int i = 0; i < num; i++) {
          if((i | i+1) == num) {
            ans.pop_back();
            ans.push_back(i);
            break;
          }
        }
      }
        
      return ans;
    }
};

File: easy/3314/main.py
class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        ans = []

        for num in nums:
            ans.append(-1);
            for i in range(num):
                if ((i | i+1) == num):
                    ans.pop()
                    ans.append(i)
                    break

        return ans
        
